### PR TITLE
Add 1.9.0 compatibility guide

### PR DESCRIPTION
**Pull request by sarahhaggarty**
This PR is a WIP.

### CODE REVIEW

File: docs/topics/compatibility-guide-19.md [hunk 1]
```diff
@@ -41,102 +41,92 @@ perspective
 > - 1.8.0: raise the warning to an error
 -->
 
-### TODO: KT-45375
-
-> **Issue**: [KT-45375](https://youtrack.jetbrains.com/issue/KT-45375)
```
[the remark this answers is not in the document]
Comment (sarahhaggarty) [reply docs/topics/compatibility-guide-19.md]: I checked with the ticket owner and this can be removed 👍

File: docs/topics/compatibility-guide-19.md [hunk 2]
```diff
@@ -148,7 +138,7 @@ perspective
 > **Incompatible change type**: source
 > 
 > **Short summary**: Kotlin will report a compile error when an explicit or implicit super call is delegated 
-> to an _abstract_ member of the superclass, even if there's a default implementation in a super interface
+> to an _abstract_ member of the superclass, even if there's a default implementation in a super interface.
```
[the remark this answers is not in the document]
Comment (sarahhaggarty) [reply docs/topics/compatibility-guide-19.md:141]: Yes I think so. If it's just one sentence you can get away with not having a period at the end. But when it is more than one sentence you really should use them consistently. Since the short summaries are sometimes one or two sentences I added the periods for every short summary for consistency.

File: docs/topics/compatibility-guide-19.md [hunk 3]
```diff
@@ -500,21 +490,51 @@ perspective
...
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: the `enableEndorsedLibs` flag is no longer supported in Gradle setup.
+>
+> **Deprecation cycle**:
+>
+> - < 1.9.0: `enableEndorsedLibs` flag is supported in Gradle setup
```
[the remark this answers is not in the document]
Comment (sarahhaggarty) [reply docs/topics/compatibility-guide-19.md:505]: It renders correctly when I previewed it locally 👍

File: docs/topics/compatibility-guide-19.md [hunk 4]
```diff
@@ -41,102 +41,92 @@ perspective
...
 >
 > **Incompatible change type**: behavioral
 >
-> **Short summary**:
+> **Short summary**: Prior to Kotlin 1.9, if you tried to override an open property from an inherited class, a compiler error is generated.
+> This is fixed in Kotlin 1.9.
 >
 > **Deprecation cycle**:
+>  * < 1.9.0: compiler error generated if overriding an open property from an inherited class
+>  * 1.9.0: no compiler error generated
```
[the remark this answers is not in the document]
Comment (sarahhaggarty) [reply docs/topics/compatibility-guide-19.md:129]: I've taken it out for now and backed up the text so I can put it in the next guide if it makes it in.

File: docs/topics/compatibility-guide-19.md [hunk 5]
```diff
@@ -41,102 +41,92 @@ perspective
...
+> **Short summary**: In Kotlin 1.9, exception messages for expression null checks do not include source code excerpts. Instead, the name of the method or field is displayed.
+> If the expression is not a method or field, there is no additional information provided in the message.
 >
 > **Deprecation cycle**:
+>  * < 1.9.0: exception messages generated by expression null checks contain source code excerpts
+>  * 1.9.0: exception messages generated by expression null checks contain method or field names only
 
 
-### TODO: KT-56545
+### Allow overriding open properties from inherited class
```
[the remark this answers is not in the document]
Comment (sarahhaggarty) [reply docs/topics/compatibility-guide-19.md:116]: I'm marking this as resolved since it's been postponed from this guide.

File: docs/topics/compatibility-guide-19.md [hunk 6]
```diff
@@ -41,102 +41,92 @@ perspective
...
 >
-> **Short summary**:
+> **Short summary**: Kotlin 1.9 prohibits super class constructor calls on functional supertypes.
 >
 > **Deprecation cycle**:
+> * 1.7.0: report a warning (or an error in progressive mode) on constructor calls on functional supertypes
+> * 1.9.0: raise the warning to an error
 
-### TODO: KT-47932
+### Prohibit cycles in annotations
```
[the remark this answers is not in the document]
Comment (sarahhaggarty) [reply docs/topics/compatibility-guide-19.md]: I've updated the text to: "Prohibit cycles in annotation parameter types". Is this OK?

File: docs/topics/compatibility-guide-19.md [hunk 7]
```diff
@@ -41,102 +41,92 @@ perspective
...
 
 > **Issue**: [KT-47932](https://youtrack.jetbrains.com/issue/KT-47932)
 >
 > **Component**: Core language
 >
 > **Incompatible change type**: source
 >
-> **Short summary**:
+> **Short summary**: Kotlin 1.9 prohibits cycles in types of annotation parameters, unless the type is an `Array` or `vararg`.
```
[the remark this answers is not in the document]
Comment (sarahhaggarty) [reply docs/topics/compatibility-guide-19.md]: I've updated to:

> Kotlin 1.9 prohibits the type of an annotation being used as one of its parameter types, either directly or indirectly. This prevents cycles from being created.
> The only exception is if the parameter type is an `Array` or `vararg` of the annotation type.

Is this better?

File: docs/topics/compatibility-guide-19.md [hunk 8]
```diff
@@ -41,102 +41,92 @@ perspective
...
+### Prohibit super class constructor call on a function interface in supertypes list
 
 > **Issue**: [KT-46344](https://youtrack.jetbrains.com/issue/KT-46344)
 >
 > **Component**: Core language
 >
 > **Incompatible change type**: source
 >
-> **Short summary**:
+> **Short summary**: Kotlin 1.9 prohibits super class constructor calls on functional supertypes.
```
[the remark this answers is not in the document]
Comment (sarahhaggarty) [reply docs/topics/compatibility-guide-19.md]: I've rewritten the title and short summary to avoid using this term. Can you check it again to see if it is clear to you?

**Review by Tapchicoma — APPROVED** [review]

File: docs/topics/compatibility-guide-19.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (ilya-g) [reply docs/topics/compatibility-guide-19.md]: Yes, but I'd rephrase the last sentence without "exception" as it may have another meaning of a runtime exception.
Like "However, using the parameter type that is an `Array` or `vararg` of the annotation type is still allowed"

File: docs/topics/compatibility-guide-19.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilya-g) [reply docs/topics/compatibility-guide-19.md:126]: ok

File: docs/topics/compatibility-guide-19.md [hunk 9]
```diff
@@ -347,15 +322,15 @@ perspective
...
+> of a compound assignment operator inside that function's body, as it currently does in other expressions inside the body of that function.
 >
 > **Deprecation cycle**:
 >
 > - 1.7.0: report a warning (or an error in the progressive mode) when a function without explicitly specified return type is
 >   called recursively in that function's body in a compound assignment operator argument
 > - 1.9.0: raise the warning to an error
 
-### Prohibit unsound calls with expected `@NotNull T` and given Kotlin generic parameter with nullable bound
+### Prohibit unsound calls with expected @NotNull T and given Kotlin generic parameter with nullable bound
```
Comment (ilya-g) [inline docs/topics/compatibility-guide-19.md:333]: Same here with quoting @
